templates/vault_main_configuration.hcl.j2

```diff
@@ -139,6 +139,16 @@ telemetry {
 license_path = "{{ vault_license_path }}"
 {% endif -%}
 
+{% if not (vault_license_reporting | bool) -%}
```

Comment:
As this is enterprise only, should we add `vault_enterprise` to this check, so it's not appended for OSS deployments?
I don't know if an OSS configuration fails if this is set.

Reply:
Thanks for the review. Gave this scenario a test. We have the same behavior as with an Enterprise version which does not support reporting:
```
[WARN]  unknown or unsupported field reporting found in configuration at /etc/vault.d/vault_main.hcl:37:1
```

Apart from the warning Vault is fully functional.